app/src/main/java/com/github/se/travelpouch/model/activity/ActivityRepositoryFirebase.kt

```diff
@@ -0,0 +1,160 @@
+package com.github.se.travelpouch.model.activity
+
+import android.util.Log
+import com.github.se.travelpouch.model.Location
+import com.google.android.gms.tasks.Task
+import com.google.firebase.Firebase
+import com.google.firebase.Timestamp
+import com.google.firebase.auth.auth
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FirebaseFirestore
+
+class ActivityRepositoryFirebase(private val db: FirebaseFirestore) : ActivityRepository {
+
+    private val collectionPath = "activities"
+
+    /**
+     * This function returns an unused unique identifier for a new event.
+     *
+     * @return (String) : an unused unique identifier
+     */
+    override fun getNewUid(): String {
+        return db.collection(collectionPath).document().id
+    }
+
+    /**
+     * The initialisation function of the repository.
+     *
+     * @param (() -> Unit) : the function to apply when the authentication goes without a trouble
+     */
+    override fun init(onSuccess: () -> Unit) {
+        Firebase.auth.addAuthStateListener {
+            if (it.currentUser != null) {
+                onSuccess()
+            }
+        }
+    }
+
```

Comment:
You can add comments for these function. It will clarify their purpose and usage.

Reply:
Done